src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -812,14 +824,19 @@ public IEnumerable<AccountHistory> GetHistory(string walletName, string accountN
 
         /// <inheritdoc />
         public AccountHistory GetHistory(HdAccount account)
+        {
+            return this.GetHistoryForAccount(account, null, null, int.MaxValue);
+        }
+      
+        protected AccountHistory GetHistoryForAccount(HdAccount account, long? prevOutputTxTime = null, int? prevOutputIndex = null, int take = int.MaxValue)
         {
             Guard.NotNull(account, nameof(account));
             FlatHistory[] items;
             lock (this.lockObject)
             {
                 // Get transactions contained in the account.
                 items = account.GetCombinedAddresses()
-                    .Where(a => a.Transactions.Any())
+                    .Where(a => take == int.MaxValue ? a.Transactions.Any() : a.AsPaginated(prevOutputTxTime, prevOutputIndex, take).Transactions.Any())
```

Comment:
Is this to implement paging? Can't we just use native `Skip` and `Take`?

Reply:
Ah wait I see now where `AsPaginated` comes from.

Reply:
Can you maybe elaborate why you are using both output Time and output Index to page?

Reply:
No not that simple unfortunately, thats how it used to work but that required loading all the transactions first which defeated the object of server side paging. Because transactions are against many addresses so we need to order by time with a previous value and then skip and take.